src/senaite/patient/api.py

```diff
@@ -202,67 +205,80 @@ def to_datetime(date_value, default=None, tzinfo=None):
     return date_value.replace(tzinfo=tzinfo)
 
 
-def to_ymd(delta):
+def to_ymd(val, default=_marker):
     """Returns a representation of a relative delta in ymd format
     """
-    if not isinstance(delta, relativedelta):
-        raise TypeError("delta parameter must be a relative_delta")
+    if not isinstance(val, relativedelta):
+        if is_ymd(val):
+            return val
+        if default is _marker:
+            raise TypeError("delta parameter must be a relative_delta or ymd")
+        return default
 
     ymd = list("ymd")
-    diff = map(str, (delta.years, delta.months, delta.days))
+    diff = map(str, (val.years, val.months, val.days))
     age = filter(lambda it: int(it[0]), zip(diff, ymd))
     return " ".join(map("".join, age))
 
 
 def is_ymd(ymd):
     """Returns whether the string represents a period in ymd format
     """
-    valid = map(lambda p: p in ymd, "ymd")
-    return any(valid)
+    values = get_years_months_days(ymd, default=None)
+    return values is not None
 
 
-def get_birth_date(age_ymd, on_date=None):
-    """Returns the birth date given an age in ymd format and the date when age
-    was recorded or current datetime if None
+def get_years_months_days(ymd, default=_marker):
+    """Returns a tuple of (years, month, days)
     """
-    on_date = to_datetime(on_date, default=datetime.now())
-
     def extract_period(val, period):
-        num = re.findall(r'(\d{1,2})'+period, val) or [0]
+        num = re.findall(r'(\d{1,2})'+period, str(val)) or [0]
         return api.to_int(num[0], default=0)
 
-    # Extract the periods
-    years = extract_period(age_ymd, "y")
-    months = extract_period(age_ymd, "m")
-    days = extract_period(age_ymd, "d")
+    years = extract_period(ymd, "y")
+    months = extract_period(ymd, "m")
+    days = extract_period(ymd, "d")
     if not any([years, months, days]):
```

Comment:
Why can't we just return `(0, 0, 0)`?
E.g. for a new born baby with 2 hours.

Reply:
yes, with latest changes it always returns `(0, 0, 0)` unless format or type are not valid